templates/terraform/extra_schema_entry/cloudiot_device_registry.go.erb

```diff
@@ -1,7 +1,9 @@
 "state_notification_config": {
-    Type:     schema.TypeMap,
```

Comment:
Isn't this a breaking change? I know it didn't super work before, but I'm not convinced that it's safe to make.

Reply:
I think this was something @paddycarver made - I vaguely remember conversation around it, but I'm pretty sure he has more context.

Reply:
I think this change particularly I'd prefer to be split out into a separate PR.

Reply:
You can't have a `TypeMap` with an `Elem` of `*schema.Resource`. It's been documented that way for a while, and if I remember right from my investigation that's even called out in the comments for this particular resource and the only reason it's preserved here is backwards compatibility concerns because it slipped through in an earlier PR.

SDKv2 has tighter checks around this and will now return an error for this unsupported behavior, so it needs to be fixed as part of upgrading to v2. If that presents undue burden, please shout and we'll see if we can figure something out.